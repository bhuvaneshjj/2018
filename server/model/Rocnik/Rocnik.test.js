'use strict';

const db = require('../../db');
const Kategorie = require('../Kategorie/Kategorie');
const Rocnik = require('./Rocnik');

beforeAll(async () => {
  await db.connect();
});

beforeEach(async () => {
  await Promise.all([Kategorie.deleteMany(), Rocnik.deleteMany()]);
});

afterAll(async () => {
  await db.disconnect();
});

it('vytvoř ročník s jednou kategorií', async () => {
  const kategorie = new Kategorie({
    typ: 'maraton',
    pohlavi: 'žena',
    vek: { min: 40, max: 49 }
  });
  await kategorie.save();

  const rocnik = new Rocnik({ rok: 2017, datum: '2017-06-10' });
  rocnik.kategorie.push({
    typ: 'maraton',
    kategorie: [kategorie.id],
    startCisla: { rozsahy: ['1-100'] },
    startovne: { predem: 150, naMiste: 200 }
  });
  await rocnik.save();

  const rocniky = await Rocnik.find({}, { _id: 0 }).populate('kategorie.kategorie', { _id: 0 });
  expect(rocniky).toMatchSnapshot();
});

it('vytvoř ročník s ubytováním', async () => {
  const kategorie1 = new Kategorie({
    typ: 'maraton',
    pohlavi: 'žena',
    vek: { min: 40, max: 49 }
  });
  await kategorie1.save();
  const kategorie2 = new Kategorie({
    typ: 'maraton',
    pohlavi: 'žena',
    vek: { min: 50, max: 59 }
  });
  await kategorie2.save();

  const rocnik = new Rocnik({ rok: 2017, datum: '2017-06-10' });
  rocnik.kategorie.push({
    typ: 'maraton',
    kategorie: [kategorie1.id, kategorie2.id],
    startCisla: { rozsahy: ['1-100'] },
    startovne: { predem: 150, naMiste: 200 }
  });
  rocnik.ubytovani.pátek = { poplatek: 50 };
  rocnik.ubytovani.sobota = { poplatek: 60 };
  await rocnik.save();

  const rocniky = await Rocnik.find({}, { _id: 0 }).populate('kategorie.kategorie', { _id: 0 });
  expect(rocniky).toMatchSnapshot();
});
