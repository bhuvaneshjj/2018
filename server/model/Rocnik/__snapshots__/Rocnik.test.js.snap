// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vytvoř ročník s jednou kategorií 1`] = `
Array [
  Object {
    "__v": 0,
    "datum": 2017-06-10T00:00:00.000Z,
    "kategorie": Array [
      Object {
        "kategorie": Array [
          Object {
            "__v": 0,
            "pohlavi": "žena",
            "typ": "maraton",
            "vek": Object {
              "max": 49,
              "min": 40,
            },
          },
        ],
        "startCisla": Object {
          "rozsahy": Array [
            "1-100",
          ],
        },
        "startovne": Object {
          "naMiste": 200,
          "predem": 150,
        },
        "typ": "maraton",
      },
    ],
    "rok": 2017,
  },
]
`;

exports[`vytvoř ročník s ubytováním 1`] = `
Array [
  Object {
    "__v": 0,
    "datum": 2017-06-10T00:00:00.000Z,
    "kategorie": Array [
      Object {
        "kategorie": Array [
          Object {
            "__v": 0,
            "pohlavi": "žena",
            "typ": "maraton",
            "vek": Object {
              "max": 49,
              "min": 40,
            },
          },
          Object {
            "__v": 0,
            "pohlavi": "žena",
            "typ": "maraton",
            "vek": Object {
              "max": 59,
              "min": 50,
            },
          },
        ],
        "startCisla": Object {
          "rozsahy": Array [
            "1-100",
          ],
        },
        "startovne": Object {
          "naMiste": 200,
          "predem": 150,
        },
        "typ": "maraton",
      },
    ],
    "rok": 2017,
    "ubytovani": Object {
      "pátek": Object {
        "poplatek": 50,
      },
      "sobota": Object {
        "poplatek": 60,
      },
    },
  },
]
`;
